src/palace/manager/core/monitor.py

```diff
@@ -813,271 +809,3 @@ class CustomListEntryWorkUpdateMonitor(CustomListEntrySweepMonitor):
-
-        CAUTION: If you override this method such that it doesn't
-        actually delete the database row, then run_once() may enter an
-        infinite loop.
-        """
-        self._db.delete(row)
-
-    def query(self):
-        return self._db.query(self.MODEL_CLASS).filter(self.where_clause)
-
-
-# ReaperMonitors that do something specific.
-
-
-class CredentialReaper(ReaperMonitor):
-    """Remove Credentials that expired more than a day ago."""
-
-    MODEL_CLASS = Credential
-    TIMESTAMP_FIELD = "expires"
-    MAX_AGE = 1
-
-
-ReaperMonitor.REGISTRY.append(CredentialReaper)
-
-
-class PatronRecordReaper(ReaperMonitor):
-    """Remove patron records that expired more than 60 days ago"""
-
-    MODEL_CLASS = Patron
-    TIMESTAMP_FIELD = "authorization_expires"
-    MAX_AGE = 60
-
-
-ReaperMonitor.REGISTRY.append(PatronRecordReaper)
-
-
-class WorkReaper(ReaperMonitor):
-    """Remove Works that have no associated LicensePools.
-
-    Unlike other reapers, no timestamp is relevant. As soon as a Work
-    loses its last LicensePool it can be removed.
-    """
-
-    MODEL_CLASS = Work
-
-    def __init__(self, *args, **kwargs):
-        search_index_client = kwargs.pop("search_index_client", None)
-        super().__init__(*args, **kwargs)
-        self.search_index_client = search_index_client or self.services.search.index()
-
-    def query(self):
-        return (
-            self._db.query(Work)
-            .outerjoin(Work.license_pools)
-            .filter(LicensePool.id == None)
-        )
-
-    def delete(self, work):
-        """Delete work from opensearch and database."""
-        work.delete(search_index=self.search_index_client)
-
-
-ReaperMonitor.REGISTRY.append(WorkReaper)
-
-
-class CollectionReaper(ReaperMonitor):
-    """Remove collections that have been marked for deletion."""
-
-    MODEL_CLASS = Collection
-
-    @property
-    def where_clause(self):
-        """A SQLAlchemy clause that identifies the database rows to be reaped."""
-        return Collection.marked_for_deletion == True
-
-    def delete(self, collection):
-        """Delete a Collection from the database.
-
-        Database deletion of a Collection might take a really long
-        time, so we call a special method that will do the deletion
-        incrementally and can pick up where it left off if there's a
-        failure.
-        """
-        collection.delete()
-
-
-ReaperMonitor.REGISTRY.append(CollectionReaper)
-
-
-class MeasurementReaper(ReaperMonitor):
-    """Remove measurements that are not the most recent"""
-
-    MODEL_CLASS = Measurement
-
-    @property
-    def where_clause(self):
-        return Measurement.is_most_recent == False
-
-    def run_once(self, *args, **kwargs):
-        rows_deleted = self.query().delete()
-        self._db.commit()
-        return TimestampData(achievements="Items deleted: %d" % rows_deleted)
-
-
-ReaperMonitor.REGISTRY.append(MeasurementReaper)
-
-
-class ScrubberMonitor(ReaperMonitor):
```

Comment:
I didn't bother migrating either of the srubber monitors, since we are not using the patron location feature. Its pretty NYPL specific. I'm going to instead just drop whats left of that code in a follow up PR to this one.

Reply:
Here is the follow up PR: https://github.com/ThePalaceProject/circulation/pull/2326 (still in progress)